### PR TITLE
Plugins: Add 'update all' button

### PR DESCRIPTION
**Pull request by johnHackworth**
Fixes https://github.com/Automattic/wp-calypso/issues/948

Currently, when users have several plugins in their "update" tab, they need to manually click in every "update" button to update them all. This PR adds a new button in the top bar to allow updating everything with a single click:

image

I'm aware that we are moving the buttons out of the top bar, but this was an easy addition and well... we can move the new button too once we tackle that issue :)

Video:
https://cloudup.com/cg6S9avpJev

How to test:
1. Go to http://calypso.dev:3000/plugins/
2. Click on updates
3. You must see an "update all" button
4. Click it and wait for all the plugins to get updated

cc/ @enejb @rickybanister

### CODE REVIEW

**Comment by rickybanister** [comment]
That button seems like an alright interim solution. I'm really excited to get around to the bulk edit / SectionHeader stuff though. That design has been kicking around forever.

The Drake message is a little confusing still, however. It implies that the updates have already happened, though you see notices that each one happens after the next. Could we introduce a half-way step where the message says 'Updating 12 Plugins' or something, and the success message appears when everything is complete?

File: client/my-sites/plugins/main.jsx [hunk 1]
```diff
@@ -633,6 +633,36 @@ export default React.createClass( {
 		return some( this.props.sites.getSelectedOrAllWithPlugins(), site => site && site.jetpack && site.canUpdateFiles );
 	},
 
+	updateAllPlugins: function() {
```
Comment (enejb) [inline client/my-sites/plugins/main.jsx]: es6 syntext :)

File: client/my-sites/plugins/main.jsx [hunk 2]
```diff
@@ -633,6 +633,36 @@ export default React.createClass( {
...
+		PluginsActions.removePluginsNotices( this.state.notices.completed.concat( this.state.notices.errors ) );
+		this.state.plugins.forEach( plugin => {
+			plugin.sites.forEach( site => PluginsActions.updatePlugin( site, site.plugin ) );
+		} );
```
Comment (enejb) [inline client/my-sites/plugins/main.jsx]: Are we updating the plugin across all sites or just the selected site?

File: client/my-sites/plugins/main.jsx [hunk 3]
```diff
@@ -633,6 +633,36 @@ export default React.createClass( {
...
+		this.recordEvent( 'Clicked Update all Plugins', true );
+	},
+
+	getNavItems() {
```
Comment (enejb) [inline client/my-sites/plugins/main.jsx]: should this function be named render instead of get? to be consistent with the rest of the functions?

**Comment by enejb** [comment]
Apart from the my minor comments I think. This is already better then the current version. So feel free to merge. After the minor items have been fixed. I think @rickybanister comments can be addressed in a different PR. That focus on just the updates state. And making sure that we display the updated info correctly as he suggested.

**Comment by enejb** [comment]
I have updated the code with the minor fixes. I think this is good to go! If you agree with the changes feel free to merge it.

**Comment by johnHackworth** [comment]
@rickybanister yeah, it's an interim solution ... the work on the section header is already ongoing at https://github.com/Automattic/wp-calypso/pull/1044 (needs some feedback, by the way ;D ). I didn't want to merge this in the same PR because the button is just 1 / 4th of the code, so once this one it's merged I can add it easily to the section header PR.

And about the half-way state .. yeah, I think you're right.  I mean, we already had this problem with single plugin updates (they disappear from the list as soon as you click on "update"), but the update all button and the bulk update in a single click make it way more obvious. We probably should introduce an "updating" state for plugin-items to improve that experience

**Pull request merged**